### PR TITLE
feat: support soft policy distributions in training loop

### PR DESCRIPTION
**Pull request by dousu**
## Summary
- replace argmax-based policy loss with a KL-divergence objective over normalized policy distributions across mixed and full precision paths
- expose the normalized policy targets on the training context and update validation metrics to consume the soft targets consistently
- configure the loss factory to provide KLDivLoss for the policy head to match the new training semantics

## Testing
- poetry run mypy src/
- poetry run python - <<'PY' ...  # soft-target training smoke test

------
https://chatgpt.com/codex/tasks/task_e_69082071555c832784f9bc2bfbea3237

### CODE REVIEW

**Comment by dousu** [comment]
@codex コンフリクトを修正して
